background.js

```diff
@@ -174,195 +176,214 @@ function backgroundAccept(sockInfo) {
-    //createNotification("onSuspend")
-    console.warn('onSuspend')
-    return
+chrome.runtime.onSuspend.addListener(function (evt) {
+	//createNotification("onSuspend")
+	console.warn('onSuspend')
+	return
 
 	// using a persistent socket now...
-    if (localOptions.optBackground) {
-        triggerKeepAwake()
-    } else {
-        if (window.app) app.stop('onsuspend')
-    }
+	if (localOptions.optBackground) {
+		triggerKeepAwake()
+	} else {
+		if (window.app) app.stop('onsuspend')
+	}
 })
-chrome.runtime.onSuspendCanceled.addListener( function(evt) {
-    //createNotification("suspendcanceled!")
-    console.warn('onSuspendCanceled')
+chrome.runtime.onSuspendCanceled.addListener(function (evt) {
+	//createNotification("suspendcanceled!")
+	console.warn('onSuspendCanceled')
 })
 
 function launch(launchData) {
 	HADEVENT = true
 
-    launchData = launchData || {}
-    //if (launchData.source == 'reload') { console.log('app was reloaded'); return }
-    if (launchData.source == 'restart') { console.log('chrome restarted'); return }
-
-    //console.log('onLaunched with launchdata',launchData)
-
-    var info = {type:'onLaunched',
-                launchData: launchData}
-    var opts = {id:'index',
-				outerBounds: { width: 410,
-							   height: 700 }
-			   }
-    //var page = 'index.html'
-    var page = 'browser.html';
-    
-    
-    //launch our app directly here
-    
-      var screenWidth = screen.availWidth;
-      var screenHeight = screen.availHeight;
-      var width = screenWidth;
-      var height = screenHeight;
-      
-      chrome.app.window.create(page, 
-      {
-        outerBounds: {
-        width: width,
-        height: height,
-        
-        }
-      },
-      function(mainWindow) {
-                                 window.mainWindow = mainWindow;
-                                 mainWindow.onClosed.addListener( window_closed )
-                                 var hiddenwin = chrome.app.window.get('hidden')
-                                 if (hiddenwin) { hiddenwin.close() }
-                                 
-			 });
-    
-
-    if (window.app) { console.log('already have webapp',app); return }
+	launchData = launchData || {}
+	//if (launchData.source == 'reload') { console.log('app was reloaded'); return }
+	if (launchData.source == 'restart') { console.log('chrome restarted'); return }
+
+	//console.log('onLaunched with launchdata',launchData)
+
+	// If launched via a url_handler, we need to grab data passed through the URL.
+	// We only want the data that comes after the base URL, specified by the ID
+	// of the url_handler.
+	var relativeUrl;
+	if (launchData.source === 'url_handler' && launchData.url && launchData.id) {
+		var regex = new RegExp('^http[s]?:\/\/' + launchData.id, 'i');
+		relativeUrl = launchData.url.replace(regex, '');
+	}
+
+	// If we identified a relative URL from the url_handler, we'll stuff
+	// it in local storage to be retrieved when the app launches.
+	if (relativeUrl) {
```

Comment:
If the URL handler was used to launch the app, we'll store the URL parameters in local storage for retrieval later. If not, we'll clear it so it doesn't take the user somewhere unexpected.